### PR TITLE
Send full view event every 5 updates under viewUpdates flag

### PR DESCRIPTION
**Pull request by maxep**
### What and why?

When the `viewUpdates` RUM feature flag is enabled, the view scope reports partial `RUMViewUpdateEvent` deltas instead of full `RUMViewEvent`s after the first event. As part of the partial view updates rollout, we need a backend correctness check that periodically compares a full view event against the aggregated deltas to catch drift or data loss in the update-aggregation pipeline, without paying the network/resource cost of comparing every single view.

A threshold of 5 updates was chosen to reconcile roughly 50% of views on Org 2 and the Demo org, giving us a much higher sampling rate for validating the update-aggregation pipeline during this phase of the rollout.

### How?

Added a `consecutiveViewUpdatesCount` counter on `RUMViewScope`, incremented each time a delta update is sent. Once it reaches `Constants.maxConsecutiveViewUpdates` (5), the scope falls back to sending a full `RUMViewEvent` and resets the counter.

### Review checklist
- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [ ] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes
- [ ] Add Objective-C interface for public APIs
- [ ] Run `make api-surface` when adding new APIs

### CODE REVIEW

File: DatadogRUM/Sources/RUMMonitor/Scopes/RUMViewScope.swift [hunk 1]
```diff
@@ -679,9 +687,12 @@ extension RUMViewScope {
         )
 
         if let event = dependencies.eventBuilder.build(from: viewEvent) {
-            if dependencies.featureFlags[.viewUpdates], let previousEvent = lastSentViewEvent {
+            if dependencies.featureFlags[.viewUpdates],
+               let previousEvent = lastSentViewEvent,
+               consecutiveViewUpdatesCount < Constants.maxConsecutiveViewUpdates {
```
[the remark this answers is not in the document]
Comment (maxep) [reply DatadogRUM/Sources/RUMMonitor/Scopes/RUMViewScope.swift:692]: Good catch, thanks. Fixed by adding an `isDeltaBaseline` flag to `RUMViewEvent.Metadata`, set whenever a full event is written under `.viewUpdates`. `RUMViewEventsFilter` now keeps every such event unconditionally instead of collapsing same-view duplicates, while legacy (non-`viewUpdates`) full-event snapshots still get collapsed as before. See e2a6dc149.

File: DatadogRUM/Sources/RUMMonitor/Scopes/RUMViewScope.swift [hunk 2]
```diff
@@ -14,6 +14,10 @@ internal class RUMViewScope: RUMScope, RUMContextProvider {
         static let minimumTimeSpentForRates = 1.0 // 1s
         /// Minimum duration of a view (1ns). Prevents negative durations and serves as placeholder value assigned when view starts.
         static let minimumTimeSpent: TimeInterval = 1e-9 // 1ns
+        /// Maximum number of consecutive `RUMViewUpdateEvent` deltas sent for a view under the `viewUpdates`
+        /// feature flag before a full `RUMViewEvent` is sent again, so the view state can be fully reconstructed
+        /// even if some update events are lost in transit. Set to 5 to reconcile 50% of views on Org 2 and the Demo org.
```
Comment (barboraplasovska) [inline DatadogRUM/Sources/RUMMonitor/Scopes/RUMViewScope.swift]: /nit Do not mention orgs here.

File: DatadogRUM/Tests/RUMMonitor/Scopes/RUMViewScope_Tests.swift [hunk 3]
```diff
@@ -426,6 +426,118 @@ class RUMViewScope_Tests: XCTestCase {
         XCTAssertEqual(lastUpdateFlags.featureFlagsInfo[flagName] as! String, flagFinalValue)
     }
 
+    // MARK: - Periodic Full View Event Resync
+
+    func testWhenConsecutiveUpdatesReachTheLimit_itResendsAFullViewEvent() throws {
```
Comment (barboraplasovska) [inline DatadogRUM/Tests/RUMMonitor/Scopes/RUMViewScope_Tests.swift:431]: /suggestion Both added tests set featureFlags: ff (viewUpdates on) for the whole test. Consider adding a test that sends `maxConsecutiveViewUpdates + 1` updates with the flag disabled and asserts no resync ever happens which isn't directly tested.

**Review by barboraplasovska — APPROVED** [review]
Looks good to me 👍🏻 , left some nits/suggestions

File: DatadogRUM/Tests/RUMMonitor/Scopes/RUMViewScope_Tests.swift (hunk 3 above)
Comment (maxep) [reply DatadogRUM/Tests/RUMMonitor/Scopes/RUMViewScope_Tests.swift:431]: Good catch, but this suite only covers the flag-on path. The flag-off path with a long update sequence is already covered in `RUMViewScopeTests.swift`, which this file mirrors and which uses the default flags (viewUpdates off). Adding a disabled-flag test here would just duplicate that.

File: DatadogRUM/Sources/RUMMonitor/Scopes/RUMViewScope.swift (hunk 2 above)
Comment (maxep) [reply DatadogRUM/Sources/RUMMonitor/Scopes/RUMViewScope.swift]: Fixed, removed the org reference.
